File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    participant.sort()
    completion.sort()
    for listadd in range(len(participant)-len(completion)):
        completion.append("0")
    for i in range(len(participant)):
        if completion[i] != participant[i]:
            answer+= participant.pop(i)
            participant.append("0")  
            
    return answer

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        int result = 0;
        HashMap<Integer,String> map = new HashMap<>();
        
        for(int i=0; i<participant.length;i++){
            int hash = participant[i].hashCode();
            result += hash;
            map.put(hash,participant[i]);
        }
        for(int i=0; i<completion.length;i++){
            result -= completion[i].hashCode();
        }
        answer = map.get(result);
        
        return answer;
    }
}